x-pack/plugins/session_view/public/components/SessionLeaderTable/index.tsx

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useState } from 'react';
+import {
+  EuiCheckbox,
+  EuiButtonIcon,
+  EuiToolTip,
+  EuiPopover,
+  EuiContextMenuPanel,
+  EuiContextMenuItem,
+} from '@elastic/eui';
+import { useStyles } from './styles';
+import { SessionViewServices } from '../../types';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import {
+  ColumnHeaderOptions,
+  ActionProps,
+  CellValueElementProps,
+  RowRenderer,
+} from '../../../../timelines/common';
+import { TGridState } from '../../../../timelines/public';
+
+export interface SessionLeaderTableProps {
+  start?: string;
+  end?: string;
+  indexNames?: string[];
+  defaultColumns?: ColumnHeaderOptions[];
+  onStateChange?: (state: TGridState) => void;
+  setRefetch?: (ref: () => void) => void;
+  itemsPerPage?: number[];
+  onExpand?: (props: ActionProps) => void;
+  onInspect?: (props: ActionProps) => void;
+  onAnalyzeSession?: (props: ActionProps) => void;
+  onOpenSessionViewer?: (props: ActionProps) => void;
+}
+
+// Not sure why the timelines plugins doesn't have a type for the
+// leading control columns props. So this is a hack to get that working for now
+type RenderLeadingControllColumnProps = {
```

Comment:
I believe x-pack/plugins/timelines/common/types/timeline/actions/index.ts has HeaderActionProps, which is what you are looking for here.

Reply:
thanks @kqualters-elastic !